featurebyte/api/target.py

```diff
@@ -0,0 +1,82 @@
+"""
+Target API object
+"""
+from __future__ import annotations
+
+from typing import Any, List, Optional
+
+from pydantic import Field
+
+from featurebyte.api.api_object import ForeignKeyMapping
+from featurebyte.api.entity import Entity
+from featurebyte.api.savable_api_object import SavableApiObject
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.models.target import TargetModel
+from featurebyte.schema.target import TargetUpdate
+
+
+class Target(SavableApiObject):
+    """
+    Target class used to represent a Target in FeatureByte.
+    """
+
+    internal_entity_ids: Optional[List[PydanticObjectId]] = Field(alias="entity_ids")
+    horizon: Optional[str]
+    blind_spot: Optional[str]
+
+    _route = "/target"
+    _update_schema_class = TargetUpdate
+
+    _list_schema = TargetModel
+    _get_schema = TargetModel
+    _list_fields = ["name", "entities"]
+    _list_foreign_keys = [
+        ForeignKeyMapping("entity_ids", Entity, "entities"),
+    ]
+
+    @property
+    def entities(self) -> List[Entity]:
+        """
+        Returns a list of entities associated with this target.
+
+        Returns
+        -------
+        List[Entity]
+        """
+        if not self.internal_entity_ids:
+            return []
+        return [Entity.get_by_id(entity_id) for entity_id in self.internal_entity_ids]
+
+    def __init__(
+        self,
+        name: str,
+        entities: Optional[List[str]] = None,
+        horizon: Optional[str] = None,
+        blind_spot: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        internal_kwargs = kwargs.copy()
+        entity_ids = None
+        if "entity_ids" in internal_kwargs:
+            entity_ids = internal_kwargs.pop("entity_ids")
+        if entities:
+            entity_ids = [Entity.get(entity_name).id for entity_name in entities]
+        super().__init__(
+            name=name,
+            entity_ids=entity_ids,
+            horizon=horizon,
+            blind_spot=blind_spot,
+            **internal_kwargs,
+        )
```

Comment:
Should we use `def create` method like what we used in other places like `Entity`, `Catalog`, `*Table`?

Reply:
hm good point, i guess `FeatureGroup` and `FeatureList` don't use `create`, but the constructor, because they can be manipulated before being saved. i don't think that'll be the case here so might be worth doing a `create` function. let me check w/ xavier

Reply:
🆗 Got it.